src/libraries/System.Private.CoreLib/src/System/Threading/WaitHandle.Windows.cs

```diff
@@ -49,20 +49,49 @@ private static unsafe int WaitForMultipleObjectsIgnoringSyncContext(IntPtr* pHan
             Thread currentThread = Thread.CurrentThread;
             currentThread.SetWaitSleepJoinState();
 
+            ulong startTime = 0;
```

Comment:
The type of this variable should be `long`. I do not think it is going to compile as is.